allennlp/semparse/contexts/table_question_context.py

```diff
@@ -76,31 +77,116 @@
         }
 
 
+class Date:
```

Comment:
Can you put this in a separate file?  Other projects want to use this also.  Also, this feels like a `DomainLanguage` construct, which belongs under `domain_languages/`, not here (to avoid circular imports in the future).  So, `domain_languages/date.py` or `domain_languages/common/date.py`.